Library/Homebrew/cmd/update-report.rb

```diff
@@ -250,7 +250,18 @@ def migrate_tap_migration
       tabs = dir.subdirs.map { |d| Tab.for_keg(Keg.new(d)) }
       next unless tabs.first.tap == tap # skip if installed formula is not from this tap.
       new_tap = Tap.fetch(new_tap_name)
-      new_tap.install unless new_tap.installed?
+      # related to formulae deduplication. If formula stays in cask, autoinstall or provide cask instructions
+      if new_tap_name == "caskroom/cask"
+        if new_tap.installed?
+          formula = Formulary.factory(new_tap_name + '/' + name)
```

Comment:
Can use string interpolation for this e.g. 
```ruby
"#{new_tap_name}/#{name}"
```

(although in this case you can hard-code `caskroom/cask`